R/module_nested_tabs.R

```diff
@@ -270,6 +299,36 @@ srv_teal_module.teal_module <- function(id,
   })
 }
 
+#' @keywords internal
+ui_validate_teal_data <- function(id) {
+  ns <- NS(id)
+  uiOutput(ns("validation_error"))
+}
+
+#' @keywords internal
+srv_validate_teal_data <- function(id, data) {
```

Comment:
It is very similar function to `srv_validate_reactive_teal_data` and has a wrong name. Please find more precise names for both functions. Also if this module is "subset" of the `srv_validate_reactive_teal_data` please consider using this function there also

Reply:
> Please find more precise names for both functions.

# 1

Looking at the purpose of `srv_validate_teal_data` I think it could be renamed to 

- `srv_verify_teal_data`
- `srv_verify_input_teal_data`
- `srv_is_empty_teal_data`

I like `srv_is_empty_teal_data` the most as it uses `.is_empty_teal_data` function inside :)

The current namespace used in `ui/srv_validate_teal_data` is called `validation_error` but we can rename to `is_empty_error`.

# 2

When it comes to `srv_validate_reactive_teal_data` I think we could rename it to
- `srv_is_error_reactive_teal_data`
- `srv_has_errors_reactive_teal_data`
- `srv_check_errors_reactive_teal_data`
- `srv_errors_in_reactive_teal_data`
- `srv_resolev_errors_reactive_teal_data`

`validate` informs the reader that some checks are performed on input `reactive(teal_data())` but I guess we only check whether the input has errors and we surface those errors with `shiny::validate` function. I think `srv_errors_in_reactive_teal_data` or `srv_check_errors_reactive_teal_data` might have more sense as we do not validate specific properties of the input but rather whether it has any errors or not.